### PR TITLE
[5.1] Added the ability to choose the 2 times for twiceDaily()

### PR DESCRIPTION
**Pull request by maartenpaauw**
Added the ability to choose the 2 times for twiceDaily()

``` php
...
$schedule->command('foo:bar')->twiceDaily('1:00', '13:00');
...
```

Defaults are 1:00 and 13:00.

### CODE REVIEW

File: src/Illuminate/Console/Scheduling/Event.php [hunk 1]
```diff
@@ -354,11 +354,21 @@ public function dailyAt($time)
     /**
      * Schedule the event to run twice daily.
      *
+     * @param null|string $first
+     * @param null|string $second
+     *
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: No newline.

File: src/Illuminate/Console/Scheduling/Event.php [hunk 2]
```diff
@@ -354,11 +354,21 @@ public function dailyAt($time)
     /**
      * Schedule the event to run twice daily.
      *
+     * @param null|string $first
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: string|null

File: src/Illuminate/Console/Scheduling/Event.php [hunk 3]
```diff
@@ -354,11 +354,21 @@ public function dailyAt($time)
...
+     * @param null|string $second
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: TWO spaces needed either side of the type

File: src/Illuminate/Console/Scheduling/Event.php [hunk 4]
```diff
@@ -354,11 +354,21 @@ public function dailyAt($time)
...
+     *
      * @return $this
      */
-    public function twiceDaily()
+    public function twiceDaily($first = '1:00', $second = '13:00')
     {
-        return $this->cron('0 1,13 * * * *');
+        $first  = explode(':', $first);
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: don't align `=`

File: src/Illuminate/Console/Scheduling/Event.php [hunk 5]
```diff
@@ -354,11 +354,21 @@ public function dailyAt($time)
...
+        $second = explode(':', $second);
+
+        $minutes = (int)count($first) == 2 ? $first[1] : 0;
+        $hours   = $first[0] . ',' . $second[0];
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: don't align `=`

File: src/Illuminate/Console/Scheduling/Event.php [hunk 6]
```diff
@@ -354,11 +354,21 @@ public function dailyAt($time)
...
      * @return $this
      */
-    public function twiceDaily()
+    public function twiceDaily($first = '1:00', $second = '13:00')
     {
-        return $this->cron('0 1,13 * * * *');
+        $first  = explode(':', $first);
+        $second = explode(':', $second);
+
+        $minutes = (int)count($first) == 2 ? $first[1] : 0;
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: space after `(int)` needed

**Comment by m44rt3np44uw** [comment]
@GrahamCampbell thanks for the feedback. Will update my code.

**Comment by taylorotwell** [comment]
If you're **always** going to be pass a `n:00` style date, why not just pass integers to make the parsing a nonissue?

File: src/Illuminate/Console/Scheduling/Event.php [hunk 7]
```diff
@@ -354,11 +354,20 @@ public function dailyAt($time)
     /**
      * Schedule the event to run twice daily.
      *
+     * @param string|null  $first
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: You need two spaces before as well as after.

File: src/Illuminate/Console/Scheduling/Event.php [hunk 8]
```diff
@@ -354,11 +354,20 @@ public function dailyAt($time)
...
+     * @param string|null  $second
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: `string|null` is actually wrong. Only `string` is allowed looking at your code.

File: src/Illuminate/Console/Scheduling/Event.php [hunk 9]
```diff
@@ -354,11 +354,20 @@ public function dailyAt($time)
     /**
      * Schedule the event to run twice daily.
      *
+     * @param  string|null  $first
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: `string|null` is actually wrong, looking at your code.

**Comment by taylorotwell** [comment]
So if you define two different minutes it only accepts the first one I guess?

**Comment by m44rt3np44uw** [comment]
@taylorotwell That's correct. Otherwise if 2 different minutes are defined the cron will run 4 times a day.

**Comment by taylorotwell** [comment]
I think this function should just accept two integers. Always use 00 for
the minutes. It's simplifies the parsing and removes that unexpected
behavior.

**Comment by m44rt3np44uw** [comment]
@taylorotwell Good thing will update the code.

File: src/Illuminate/Console/Scheduling/Event.php [hunk 10]
```diff
@@ -354,11 +354,16 @@ public function dailyAt($time)
...
      *
+     * @param  int  $first
+     * @param  int  $second
      * @return $this
      */
-    public function twiceDaily()
+    public function twiceDaily($first = 1, $second = 13)
     {
-        return $this->cron('0 1,13 * * * *');
+        $hours = $first . ',' . $second;
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: No spaces for concatenation please.

**Comment by JosephSilber** [comment]
Maybe also add a `twiceDailyAt` alias, similar to `dailyAt`.

**Pull request merged**